src/reducers/entities/users.ts

```diff
@@ -67,6 +67,14 @@ function addProfileToSet(state: RelationOneToMany<Team, UserProfile>, action: Ge
     };
 }
 
+function removeProfileFromTeams(state: RelationOneToMany<Team, UserProfile>, action: GenericAction) {
+    const newState = {...state};
+    Object.keys(state).forEach((key) => {
```

Comment:
Maybe consider having an additional variable to keep track if anything actually change in the state to avoid returning a new state if nothing was changed.